akka-persistence-typed/src/test/java/akka/persistence/typed/javadsl/PersistentActorJavaDslTest.java

```diff
@@ -160,184 +170,116 @@ public int hashCode() {
-    return counter(persistenceId, probe, loggingProbe, (s, i, l) -> false, (e) -> Collections.emptySet(), snapshotProbe, new NoOpEventAdapter<>());
+  private Behavior<Command> counter(String persistenceId) {
+    return Behaviors.setup(ctx -> new CounterBehavior(persistenceId, ctx));
   }
 
-  private Behavior<Command> counter(String persistenceId,
-                                    ActorRef<Pair<State, Incremented>> probe,
-                                    Function<Incremented, Set<String>> tagger) {
-    ActorRef<String> loggingProbe = TestProbe.create(String.class, testKit.system()).ref();
-    ActorRef<Optional<Throwable>> snapshotProbe = TestProbe.<Optional<Throwable>>create(testKit.system()).ref();
-    return counter(persistenceId, probe, loggingProbe, (s, i, l) -> false, tagger, snapshotProbe, new NoOpEventAdapter<>());
-  }
+  private static class CounterBehavior extends PersistentBehavior<Command, Incremented, State> {
+    private final ActorContext<Command> ctx;
 
- private Behavior<Command> counter(String persistenceId,
-                                   ActorRef<Pair<State, Incremented>> probe,
-                                   EventAdapter<Incremented, ?> transformer) {
-    ActorRef<String> loggingProbe = TestProbe.create(String.class, testKit.system()).ref();
-    ActorRef<Optional<Throwable>> snapshotProbe = TestProbe.<Optional<Throwable>>create(testKit.system()).ref();
-    return counter(persistenceId, probe, loggingProbe, (s, i, l) -> false, e -> Collections.emptySet(), snapshotProbe, transformer);
-  }
+    CounterBehavior(String persistentId, ActorContext<Command> ctx) {
+      super(persistentId, SupervisorStrategy.restartWithBackoff(Duration.ofMillis(1), Duration.ofMillis(5), 0.1));
+      this.ctx = ctx;
+    }
 
-  private Behavior<Command> counter(String persistenceId) {
-    return counter(persistenceId,
-      TestProbe.<Pair<State, Incremented>>create(testKit.system()).ref(),
-      TestProbe.<String>create(testKit.system()).ref(),
-      (s, i, l) -> false,
-      (i) -> Collections.emptySet(),
-      TestProbe.<Optional<Throwable>>create(testKit.system()).ref(),
-      new NoOpEventAdapter<>()
-    );
-  }
+    @Override
+    public CommandHandler<Command, Incremented, State> commandHandler() {
+      return commandHandlerBuilder(State.class)
+          .matchCommand(Increment.class, this::increment)
+          .matchCommand(GetValue.class, this::getValue)
+          .matchCommand(IncrementLater.class, this::incrementLater)
+          .matchCommand(DelayFinished.class, this::delayFinished)
+          .matchCommand(Increment100OnTimeout.class, this::increment100OnTimeout)
+          .matchCommand(Timeout.class, this::timeout)
+          .matchCommand(EmptyEventsListAndThenLog.class, this::emptyEventsListAndThenLog)
+          .matchCommand(StopThenLog.class, this::stopThenLog)
+          .matchCommand(IncrementTwiceAndLog.class, this::incrementTwiceAndLog)
```

Comment:
Separating command handlers into individual methods makes it more obvious what is purely dispatch logic and what is relevant to the domain. It also reduces some of the syntax soup that might be unfamiliar to some Java developers who haven't used lambdas much. Method handles might still be new, but maybe a little easier to intuit.

Reply:
added this to the style guide ticket https://github.com/akka/akka/issues/22805